drivers/counter/counter_native_posix.c

```diff
@@ -0,0 +1,156 @@
+/*
+ * Copyright (c) 2020, Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <device.h>
+#include <drivers/counter.h>
+#include <soc.h>
+#include <hw_counter.h>
+
+#define DT_COUNTER_LABEL counter0
+#define DRIVER_CONFIG_INFO_FLAGS (COUNTER_CONFIG_INFO_COUNT_UP)
+#define DRIVER_CONFIG_INFO_CHANNELS 1
+
+#define COUNTER_PERIOD (USEC_PER_SEC / CONFIG_COUNTER_NATIVE_POSIX_FREQUENCY)
+#define TOP_VALUE (UINT_MAX)
+
+static struct counter_alarm_cfg pending_alarm;
+static bool is_alarm_pending;
+static struct device *device;
+
+static void counter_isr(void *arg)
+{
+	ARG_UNUSED(arg);
+	uint32_t current_value = hw_counter_get_value();
+
+	if (is_alarm_pending) {
+		is_alarm_pending = false;
+		pending_alarm.callback(device, 0, current_value,
+				       pending_alarm.user_data);
+	}
+}
+
+static int ctr_init(struct device *dev)
+{
+	device = dev;
+	is_alarm_pending = false;
+
+	IRQ_CONNECT(COUNTER_EVENT_IRQ, 2, counter_isr, 0, NULL);
```

Comment:
IRQ_CONNECT(COUNTER_EVENT_IRQ, 2, counter_isr, 0, 0);
NULL => 0 (that last field is flags (causes a compile warning))

Reply:
Fixed. Moreover I have #defined the other arguments to get rid of magic numbers.